### PR TITLE
MySQL: INSERT ... ON DUPLICATE KEY UPDATE

### PR DESCRIPTION
**Pull request by kornrunner**
Is there a plan for supporting `INSERT INTO ... ON DUPLICATE KEY UPDATE` variaton of `INSERT` for MySQL? I'd attempt to work it out and send a pull request if there's any interest.

### CODE REVIEW

**Comment by kornrunner** [comment]
I've added the functionality, implemented three new methods:
`colOnUpdate`
`colsOnUpdate` and
`setOnUpdate` - to match existing functionality.

I've also had to hold column values in different array for that part of query, and changed bind names for those columns - to avoid mixing with original insert values... Not the happiest solution I think, do you have some other suggestion?

Also, I'm not really satisfied with method names and with the tests, but it's something... Please let me know what do you think. Thanks!

File: src/Mysql/Insert.php [hunk 1]
```diff
@@ -80,4 +90,121 @@ public function delayed($enable = true)
...
+
+    /**
+     *
+     * Sets one column value placeholder in ON DUPLICATE KEY UPDATE section;
+     * if an optional second parameter is passed, that value is bound to the
+     * placeholder.
+     *
+     * @param string $col The column name.
+     *
+     * @param mixed $val Optional: a value to bind to the placeholder.
```
Comment (staabm) [inline src/Mysql/Insert.php]: Not existent

Comment (kornrunner) [inline src/Mysql/Insert.php]: This was supposed to be equivalent of `col` method from https://github.com/auraphp/Aura.SqlQuery/blob/develop-2/src/Common/Insert.php#L131 and code is almost the same as that method - I've left the same phpdoc as well (it's using `func_get_args`)

Comment (staabm) [inline src/Mysql/Insert.php]: I see. per phpdocumentor this should read `mixed,...` instead of `mixed`

File: src/Common/Insert.php [hunk 2]
```diff
@@ -123,7 +123,7 @@ public function getLastInsertIdName($col)
      *
      * @param string $col The column name.
      *
-     * @param mixed  $val Optional: a value to bind to the placeholder.
+     * @param mixed,...  $val Optional: a value to bind to the placeholder.
```
Comment (harikt) [inline src/Common/Insert.php:126]: oh I wasn't aware we need `,...` .

**Comment by harikt** [comment]
Thank you @kornrunner for your work. @pmjones will look and merge. I am ok with this for I don't have a use case currently.

Question I have in mind is 
1. Hope it will not be having any problem passing the query to PDO. 

May be we need to document this if @pmjones is satisfied before merge. So you can wait to hear him before your work :+1:  .

**Comment by pmjones** [comment]
@kornrunner I changed the method names to `onDuplicateKeyUpdate()`, `onDuplicateKeyUpdateCol()`, and `onDuplicateKeyUpdateCols()` (for set, col, and cols, respectively), and used suffixes instead of prefixes on the modified bind placeholder names. Thanks!

**Comment by kornrunner** [comment]
@pmjones Thank you very much!

**Pull request merged**